### PR TITLE
Mis-parsing GraphQL identifiers when it starts with keywords (fix #20)

### LINKED ISSUE #20: Mis-Parsing of Columns starting with keywords
Bug surfaced in https://github.com/hasura/graphql-engine/issues/3597

PR: https://github.com/hasura/graphql-parser-hs/pull/21

TODO:

* [x] - Fix issue
* [x] - Reference new commit from engine
* [x] - Verify fix manually
* [x] - Add python test to replicate issue
* [x] - Run python test suite
* [ ] - Merge PR to parser
* [ ] - Reference commit in engine

### PR DESCRIPTION
**Pull request by sordina**
Related Issues:

* Issue https://github.com/hasura/graphql-parser-hs/issues/20
* Issue https://github.com/hasura/graphql-engine/issues/3597

Ensures that primitive identifiers end rather than appear as a prefix of a broader identifier, ala `nullColumnTwo`.

Unit-tests added to ensure that the issue does not re-occur.

Currently, the implementation may be inefficient due to lookup of char-class:

```haskell
identChars :: String
identChars = ['a'..'z'] ++ ['A'..'Z'] ++ ['0'..'9'] ++ "_"
```

This could also be wrong!

Ideally lexing and parsing will be separated and lexing can be compared to the syntax spec more easily.

### CODE REVIEW

**Comment by sordina** [comment]
@ecthiender can you review? Once this has been passed I'll be able to reference from graphql-engine.

File: test/primitiveColumns.hs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE OverloadedStrings #-}
```
Comment (ecthiender) [inline test/primitiveColumns.hs]: Name of the file should be `PrimitiveColumns`.

Comment (ecthiender) [reply test/primitiveColumns.hs]: Also, I think we should call it something else. It is not really related to columns, but primitive values of GraphQL.

File: package.yaml [hunk 2]
```diff
@@ -78,6 +78,7 @@ benchmarks:
 tests:
   graphql-parser-test:
     main:                Spec.hs
+    other-modules:       PrimitiveColumns
```
Comment (ecthiender) [inline package.yaml]: In the hspec format, I don't think we have to include them like this. Do you think this is necessary?

File: src/Language/GraphQL/Draft/Parser.hs [hunk 3]
```diff
@@ -411,6 +414,20 @@ tok :: AT.Parser a -> AT.Parser a
 tok p = p <* whiteSpace
 {-# INLINE tok #-}
 
+literal :: AT.Parser a -> AT.Parser a
```
Comment (ecthiender) [inline src/Language/GraphQL/Draft/Parser.hs]: I think we can add some comment describing why needed this and why we have a separate `tok` function as well. @sordina what do you think?

Comment (sordina) [reply src/Language/GraphQL/Draft/Parser.hs]: Sounds good!

**Pull request merged**